graphql/introspection/introspection.go

```diff
@@ -123,6 +123,15 @@ var skipDirective = Directive{
 	},
 }
 
+var typeAsOptionalDirective = Directive{
+	Description: "Client-side-only directive that instructs the type generator to mark this field as optional. This is useful for making the generated types compliant with Troy persistence schema.",
+	Locations: []DirectiveLocation{
+		FIELD,
```

Comment:
Just double-checking, this means we can't apply this directive to an entire fragment

Reply:
yep! it doesn't really make sense in that scenario